gsa/src/web/pages/extras/cvsscalculatorpage.js

```diff
@@ -359,21 +350,388 @@ const CvssCalculator = ({gmp, onInteraction, ...props}) => {
+
+    if (
+      isDefined(location) &&
+      isDefined(location.query) &&
+      isDefined(location.query.cvssVector)
+    ) {
+      const {cvssVector} = location.query;
+      setState(vals => ({...vals, cvssVector, userVector: cvssVector}));
+      handleVectorChange();
+
+      let newScore = calculateV3Score({
+        attackVector,
+        attackComplexity,
+        privilegesRequired,
+        userInteraction,
+        scope,
+        confidentialityImpact,
+        integrityImpact,
+        availabilityImpact,
+      });
+      setState(vals => ({...vals, cvssScore: newScore}));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    handleVectorChange();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.userVector]);
+
+  const calculateVector = newVector => {
+    const {
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+    } = state;
+
+    const cvssVector = parseCvssV3BaseVector({
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+      ...newVector,
+    });
+
+    setState(vals => ({
+      ...vals,
+      ...newVector,
+      cvssVector,
+      userVector: cvssVector,
+    }));
+
+    let newScore = calculateV3Score({
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+    });
+    setState(vals => ({...vals, cvssScore: newScore}));
+  };
+
+  const handleInteraction = () => {
+    renewSession();
+  };
+
+  const handleMetricsChange = (value, name) => {
+    handleInteraction();
+
+    calculateVector({[name]: value});
+  };
+
+  const handleInputChange = (value, name) => {
+    setState(vals => ({...vals, [name]: value}));
+  };
+
+  const handleVectorChange = () => {
+    const {userVector} = state;
+
+    handleInteraction();
+
+    const cvssValues = parseCvssV3BaseFromVector(userVector);
+    const {
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+    } = cvssValues;
+
+    if (
+      isDefined(attackVector) &&
+      isDefined(attackComplexity) &&
+      isDefined(privilegesRequired) &&
+      isDefined(userInteraction) &&
+      isDefined(scope) &&
+      isDefined(confidentialityImpact) &&
+      isDefined(integrityImpact) &&
+      isDefined(availabilityImpact)
+    ) {
+      /* only override cvss values and vector if user vector has valid input */
+
+      setState(vals => ({...vals, ...cvssValues, cvssVector: userVector}));
+
+      let newScore = calculateV3Score({
```

Comment:
Any reason for not using const here?

Reply:
Already changed it ... but it is not commited ...